### PR TITLE
PR-Lint: Implement commit count check

### PR DESCRIPTION
**Pull request by aliyevakhalida**
- Implemented commit count check. Each PR should have at most 10 (default count, can be customized with input) commits
- Excluded release and sync PRs

### CODE REVIEW

**Review by aytajde — DISMISSED** [review]

File: .github/workflows/cicd_pipeline_android.yaml
Comment (kerimovscreations) [inline .github/workflows/cicd_pipeline_android.yaml:1]: please do not modify this file. This PR is not intended for this file

File: pr-format/scripts/validate_commit_count.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+BRANCH_NAME="$1"
+PR_COMMIT_COUNT="$2"
+MAX_COMMIT_COUNT="$3"
+GITHUB_ENV="$4"
+
+SYNC_REGEX="^tech/sync"
```
Comment (kerimovscreations) [inline pr-format/scripts/validate_commit_count.sh:8]: I suggest to include `tech` branches for these changes

File: pr-format/scripts/validate_commit_count.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+RELEASE_REGEX="^release/"
+
+if [[ $BRANCH_NAME =~ $SYNC_REGEX || $BRANCH_NAME =~ $RELEASE_REGEX ]]; then
+    IS_VALID=true
+    TEXT="Number of commits is less than the upper limit of ${MAX_COMMIT_COUNT} commits"
```
Comment (kerimovscreations) [inline pr-format/scripts/validate_commit_count.sh:13]: Please use proper message for these exception branches. You can mention within brackets `()`, or by changing wording.

**Review by kerimovscreations — CHANGES_REQUESTED** [review]

File: .github/workflows/cicd_pipeline_android.yaml
Comment (aliyevakhalida) [reply .github/workflows/cicd_pipeline_android.yaml:1]: Right, amended

File: pr-format/scripts/validate_commit_count.sh (hunk 1 above)
Comment (aliyevakhalida) [reply pr-format/scripts/validate_commit_count.sh:8]: @kerimovscreations, you mean to exclude all tech PRs (not only tech/sync)?

File: pr-format/scripts/validate_commit_count.sh (hunk 2 above)
Comment (aliyevakhalida) [reply pr-format/scripts/validate_commit_count.sh:13]: I overlooked that one 😅 coming up with a proper message 📣

Comment (aliyevakhalida) [reply pr-format/scripts/validate_commit_count.sh]: Amended

**Review by kerimovscreations — APPROVED** [review]

**Review by realhuseyn — APPROVED** [review]

**Pull request merged**